helix-core/src/lib.rs

```diff
@@ -45,7 +45,12 @@ pub fn find_first_non_whitespace_char(line: RopeSlice) -> Option<usize> {
 /// * Top-most folder containing a root marker if not git repository detected
 /// * Current working directory as fallback
 pub fn find_root(root: Option<&str>, root_markers: &[String]) -> Option<std::path::PathBuf> {
+    find_root_impl(root, root_markers).first().cloned()
+}
+
+fn find_root_impl(root: Option<&str>, root_markers: &[String]) -> Vec<std::path::PathBuf> {
     let current_dir = std::env::current_dir().expect("unable to determine current directory");
```

Comment:
Why the change here if you end up ignoring everything except the first element anyway?

Reply:
I needed `find_root_impl` to return a list of all directories that match `root` up to `.git` for `local_config_dirs()`

Reply:
e.g.
```
.
└── project/
    ├── .git
    ├── .helix
    └── foo/
        └── .helix/
            └── bar/
                └── .helix
```